File: src/templates/CSharpTemplate/Options.cs
#region License
// <copyright file="Options.cs" company="Your name here">
//   Copyright 2013 Your name here
// </copyright>
//
// [License Body Here]
#endregion

namespace CSharpTemplate
{
    #region Using Directives
    using CommandLine;
    using CommandLine.Text;
    #endregion

    internal class Options
    {
        [Option('t', "text", Required = true, HelpText = "text value here")]
        public string TextValue { get; set; }

        [Option('n', "numeric", HelpText = "numeric value here")]
        public double NumericValue { get; set; }

        [Option('b', "bool", HelpText = "on|off switch here")]
        public bool BooleanValue { get; set; }

        [HelpOption]
        public string GetUsage()
        {
            return HelpText.AutoBuild(this, current => HelpText.DefaultParsingErrorsHandler(this, current));
        }
    }
}

File: src/templates/VBNetTemplate/Options.vb
#Region "License"
'<copyright file="Options.vb" company="Your name here">
'   Copyright 2013 Your name here
' </copyright>
'
' [License Body Here]
#End Region
#Region "Using Directives"
Imports CommandLine
Imports CommandLine.Text
#End Region

Friend Class Options

    <[Option]("t"c, "text", Required:=True, HelpText:="text value here")>
    Public Property TextValue As String

    <[Option]("n"c, "numeric", HelpText:="numeric value here")>
    Public Property NumericValue As Double = 0

    <[Option]("b"c, "bool", HelpText:="on|off switch here")>
    Public Property BooleanValue As Boolean

    <HelpOption()>
    Public Function GetUsage() As String

        Return HelpText.AutoBuild(Me, Sub(current As HelpText) HelpText.DefaultParsingErrorsHandler(Me, current))

    End Function
End Class